File: pset2/caesar/caesar.c
/*
 * Caesar
 * CS50 Problem Set 2
 *
 * Encrypts text using Caesar cipher
 */
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void encrypt(string plain, int key);

int main(int argc, string argv[])
{
    // Error if invalid number of args
    if (argc != 2)
    {
        printf("Usage: caesar KEY\n");
        return 1;
    }

    // Initialize key
    int key = atoi(argv[1]);

    // Prompts user for input
    string plaintext = get_string("plaintext: ");

    // Encrypts
    printf("ciphertext: ");
    encrypt(plaintext, key);

    return 0;
}

void encrypt(string plain, int key)
{
    // Iterate over the string
    for (int i = 0, length = strlen(plain); i < length; i++)
    {
        // Uppercase
        if (isupper(plain[i]))
        {
            printf("%c", (plain[i] - 'A' + key) % 26 + 'A');
        }
        // Lowercase
        else if (islower(plain[i]))
        {
            printf("%c", (plain[i] - 'a' + key) % 26 + 'a');
        }
        // Non alphabetic characters
        else
        {
            printf("%c", plain[i]);
        }
    }
    printf("\n");
    return;
}

File: pset6/caesar/caesar.py
"""
Caesar
CS50 Problem Set 6

Encrypts text using Caesar cipher
"""
import sys
import string


UPPERCASE = string.ascii_uppercase
LOWERCASE = string.ascii_lowercase


def main():
    # Error if invalid number of args
    if not len(sys.argv) is 2:
        print("Usage: caesar KEY")
        exit(1)

    # Check if key is int
    try:
        key = int(sys.argv[1])
    except ValueError:
        print("Key must be integer")
        exit(1)

    # Prompts user for input
    plaintext = input("plaintext: ")

    # Encrypts
    print("ciphertext:", encrypt(plaintext, key))


def encrypt(plain, key):
    # String to store cipher text
    cipher = ""

    # Iterate over string
    for i in plain:
        # Uppercase
        if i.isupper():
            shift = (UPPERCASE.index(i) + key) % len(UPPERCASE)
            cipher += UPPERCASE[shift]
        # Lowercase
        elif i.islower():
            shift = (LOWERCASE.index(i) + key) % len(LOWERCASE)
            cipher += LOWERCASE[shift]
        # Non alphabetic characters
        else:
            cipher += i

    return cipher


if __name__ == "__main__":
    main()
